.github/workflows/master.yml

```diff
@@ -42,13 +46,17 @@ jobs:
       - name: Build and test Conformance-Gateway
         run: |
           cd Conformance-Gateway
-          mvn verify -U -B -V --fail-at-end
+          mvn package -U -B -V --fail-at-end
+
+      - name: Set Sonar settings for PR build
+        if: ${{ (github.event_name == 'pull_request') }}
+        run: |
+          echo "SONAR_PARAMS=-Dsonar.pullrequest.base=dev -Dsonar.pullrequest.key=${{github.event.pull_request.number}} -Dsonar.pullrequest.branch=$BRANCH_NAME" >> $GITHUB_ENV
 
       - name: Run SonarCloud analysis
         env:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }} # SonarCloud
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # SonarCloud
-          BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
         run: |
           cd Conformance-Gateway
-          mvn sonar:sonar -B -Dsonar.pullrequest.base=dev -Dsonar.pullrequest.key=${{github.event.pull_request.number}} -Dsonar.pullrequest.branch=$BRANCH_NAME
+          mvn sonar:sonar -B ${{env.SONAR_PARAMS}}
```

Comment:
I think this would break and cause command line injection with branches that have spaces in them (the original version probably did too).

Maybe something like this

```shell
   if [ "${{github.event_name}}" == "pull_request" ]; then
        mvn sonar:sonar -B -Dsonar.pullrequest.base=dev -Dsonar.pullrequest.key="${{github.event.pull_request.number}}" -Dsonar.pullrequest.branch="$BRANCH_NAME"
   else
        mvn sonar:sonar -B -Dsonar.branch.name="$BRANCH_NAME"
   fi

```

And then remove `Set Sonar settings for PR build`?

Reply:
I cannot produce any issues with the current setup and the proposed code might be safe. #124  and #125 were the attempts in case someone wants to give at a go.